pkg/cmd/server/origin/master_config.go

```diff
@@ -127,6 +127,17 @@ type MasterConfig struct {
 // BuildMasterConfig builds and returns the OpenShift master configuration based on the
 // provided options
 func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
+	var kubeClient *etcdclient.Client
+	var err error
+	var serviceAccountTokenGetter serviceaccount.ServiceAccountTokenGetter
+
+	if options.KubernetesEtcdClientInfo != nil {
+		kubeClient, err = etcd.EtcdClient(*options.KubernetesEtcdClientInfo)
```

Comment:
and I would actually do this after constructing the originEtcdClient, and default it to originEtcdClient if KubernetesEtcdClientInfo is nil